src/fluree/server/handlers/shared.clj

```diff
@@ -27,8 +31,38 @@
              (throw (ex-info "Error in ledger handler"
                              {:response
                               {:status status#
-                               :body   (assoc error# :message msg#)}})))))
+                               :body   (assoc error# :message msg#)}}
+                             e#)))))
        (catch Throwable t#
          (throw (ex-info "Error in ledger handler"
                          {:response {:status 500
-                                     :body   {:error (ex-message t#)}}}))))))
+                                     :body   {:error (ex-message t#)}}}
+                         t#))))))
+
+(defn with-header
+  [response header value]
+  (update response :headers assoc header value))
+
+(defn with-time-header
+  [response time]
+  (with-header response "x-fdb-time" time))
+
+(defn with-fuel-header
+  [response fuel]
+  (with-header response "x-fdb-fuel" (str fuel)))
+
+(defn base64-encode
+  [^String s]
+  (.encodeToString (Base64/getEncoder) (.getBytes s)))
+
+(defn with-policy-header
+  [response policy]
+  (let [encoded-policy (-> policy json/stringify base64-encode)]
```

Comment:
Is it worth base64 encoding this? It makes the response larger and forces the end user to decode it properly. I think it would be simpler to just use `json/stringify` without b64 encoding it.

Reply:
base64 encoding this is the most reliable way to ensure consumers can reliably reconstruct the data. 

http allows multiple headers with the same key, and `x-fdb-policy: foo`, `x-fdb-policy: bar`, and `x-fdb-policy: baz` is interpreted as `x-fdb-policy: foo,bar,baz`, and vice versa. 

i don't want commas to appear in a header payload because commas have special meaning in a header. that's why stringified json isn't enough.